Category.agda

```diff
@@ -0,0 +1,121 @@
+{-# OPTIONS --cubical #-}
+module Category where
+
+open import PathPrelude
+
+-- Functor
+record Func {ℓ} : Set (ℓ-suc ℓ) where
+  field
+    map      : Set ℓ → Set ℓ
+    fmap     : {A B : Set ℓ} → (A → B) → map A → map B
+    presId   : (A : Set ℓ) → fmap (idFun A) ≡ idFun (map A)
+    presComp : {A B C : Set ℓ} → (f : A → B) → (g : B → C) →
+                 fmap (g ∘ f) ≡ (fmap g) ∘ (fmap f)
+
+-- Functor Path
+record Func≡ {ℓ} : Set (ℓ-suc ℓ) where
+  field F G : Func {ℓ}
+  module L = Func F
+  module R = Func G
+
+  field
+    map≡      : (A : Set ℓ) →
+      Path
+      (L.map A)        (R.map A)
+    fmap≡     : {A B : Set ℓ} → (f : A → B) →
+      PathP (λ i → map≡ A i → map≡ B i)
+      (L.fmap f)       (R.fmap f)
+    presId≡   : (A : Set ℓ) →
+      PathP (λ i → fmap≡ (idFun A) i ≡ idFun (map≡ A i))
+      (L.presId A)     (R.presId A)
+    presComp≡ : {A B C : Set ℓ} → (f : A → B) → (g : B → C) →
+      PathP (λ i → fmap≡ (g ∘ f) i ≡ (fmap≡ g) i ∘ (fmap≡ f) i)
+      (L.presComp f g) (R.presComp f g)
+
+  path : F ≡ G
+  path i = record { map      = λ A   → map≡      A   i
+                  ; fmap     = λ f   → fmap≡     f   i
+                  ; presId   = λ A   → presId≡   A   i
+                  ; presComp = λ f g → presComp≡ f g i }
+
+idᶠ : ∀{ℓ} → Func {ℓ}
+idᶠ = record { map      = idFun (Set _)
+             ; fmap     = λ {A B} → idFun (A → B)
+             ; presId   = λ _ → refl
+             ; presComp = λ _ _ → refl }
+
+-- Composition of Functor
+infixr 45 _∘ᶠ_
+_∘ᶠ_ : ∀{ℓ} → Func {ℓ} → Func {ℓ} → Func {ℓ}
+G ∘ᶠ F = record
+  { map      = G.map ∘ F.map
+  ; fmap     = G.fmap ∘ F.fmap
+  ; presId   = λ A → trans (cong G.fmap (F.presId A)) (G.presId _)
+  ; presComp = λ f g → trans (cong G.fmap (F.presComp f g)) (G.presComp _ _)
+  } where module F = Func F
+          module G = Func G
+
+idᶠLeft : ∀{ℓ}{F : Func {ℓ}} → idᶠ ∘ᶠ F ≡ F
+idᶠLeft {ℓ} {F} = Func≡.path r where
+  module L = Func (idᶠ ∘ᶠ F)
+  module R = Func F
+  map≡ : L.map ≡ R.map
+  map≡ = funExt (λ _ → refl)
+  r = record { F         = idᶠ ∘ᶠ F
+             ; G         = F
+             ; map≡      = λ A → refl
+             ; fmap≡     = λ f → funExt λ _ → refl
+             ; presId≡   = λ A → {!!}
+             ; presComp≡ = λ f g → {!!}
+             }
```

Comment:
Please ignore Module L, R and map≡, these were legacy code before I had an idea to created Func≡

If we give a path directly, (unless having ugly duplication), we cannot construct fmap in record {... ; ... ; ...}  without auxiliary in where clouse. This is because fmap need to refer back to map, (this dependency is identified in Type of Func≡.fmap≡). The same apply to presId≡ and presComp≡ which need to refer back to and map≡ and fmap≡.

By creating Func≡ we move these cumbersome setting in to one place in uniform manner. Later if we need to construct path for Func again it would be shorter and easier to read.